src/instructlab/model/evaluate.py

```diff
@@ -0,0 +1,504 @@
+# Standard
+import os
+import subprocess
+import time
+
+# Third Party
+from instructlab.eval.evaluator import Evaluator
+from instructlab.eval.mmlu import MMLUBranchEvaluator, MMLUEvaluator
+from instructlab.eval.mt_bench import MTBenchBranchEvaluator, MTBenchEvaluator
+import click
+
+BENCHMARK_TO_CLASS_MAP = {
+    "mmlu": MMLUEvaluator,
+    "mmlu_branch": MMLUBranchEvaluator,
+    "mt_bench": MTBenchEvaluator,
+    "mt_bench_branch": MTBenchBranchEvaluator,
+}
+
+# TODO: Remove after proper vLLM integration
+# pylint: disable=consider-using-with
+
+
+def get_evaluator(
+    model,
+    base_model,
+    benchmark,
+    judge_model,
+    output_dir,
+    max_workers,
+    taxonomy_path,
+    branch,
+    base_branch,
+    few_shots,
+    batch_size,
+    sdg_path,
+) -> Evaluator:
+    """takes in arguments from the CLI and uses 'benchmark' to validate other arguments
+    if all needed configuration is present, returns the appropriate Evaluator class for the benchmark
+    otherwise raises an exception for the missing values
+    """
+
+    # ensure skills benchmarks have proper arguments if selected
+    if benchmark in ["mt_bench", "mt_bench_branch"]:
+        required_args = [
+            model,
+            judge_model,
+            output_dir,
+            max_workers,
+        ]
+        required_arg_names = [
+            "model",
+            "judge-model",
+        ]
+        if benchmark == "mt_bench_branch":
+            required_args.append(taxonomy_path)
+            required_args.append(branch)
+            required_args.append(base_branch)
+            required_args.append(base_model)
+            required_arg_names.append("taxonomy-path")
+            required_arg_names.append("branch")
+            required_arg_names.append("base-branch")
+            required_arg_names.append("base-model")
+        if None in required_args:
+            click.secho(
+                f"Benchmark {benchmark} requires the following args to be set: {required_arg_names}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        evaluator_class = BENCHMARK_TO_CLASS_MAP[benchmark]
+        if benchmark == "mt_bench":
+            return evaluator_class("test_model", "judge_model", output_dir, max_workers)
```

Comment:
Should `test_model` be `model`? Applies here and other places too

Reply:
In this case test_model is simply the name referenced by mt_bench to call the model being served.  It could be anything so we just want to pick something recognizable.  And calling the model you are serving "model" seems a little under descriptive.

Reply:
Sounds good to me! Wanted to ensure it wasn't missed with the other renames to 'model'